test/packages/good_v2/data_stream/routing_rules/routing_rules.yml

```diff
@@ -0,0 +1,35 @@
+routing_rules:
+  # "Local" routing rules are included under this current dataset, not a special case
+  - dataset: routing_rules
+    rules: # Route error logs to `routing_rules.error` when they're sourced from an error logfile
+    - dataset: routing_rules.error
```

Comment:
It may be confusing to have two `dataset` attributes at different levels, could we be more specific about what each of them represent?

Reply:
This is related to the question I posted here https://github.com/elastic/package-spec/pull/535#discussion_r1234204623

From what I understand, I think the first `dataset` (at line 3) could be interpreted as "source dataset". Is that right? 

If so, maybe it could be renamed both fields to be `source_dataset` and `target_dataset` ?

Reply:
Options that I could think of @juliaElastic @jsoriano :
- source_dataset and target_dataset
```yaml
# "Local" routing rules are included under this current dataset, not a special case
- source_dataset:  nginx
  # Route error logs to `nginx.error` when they're sourced from an error logfile
  - target_dataset: nginx.error
    if: "ctx?.file?.path?.contains('/var/log/nginx/error')"
    namespace:
      - {{labels.data_stream.namespace}}
      - default

  # Route access logs to `nginx.access` when they're sourced from an access logfile
  - target_dataset: nginx.access
    if: "ctx?.file?.path?.contains('/var/log/nginx/access')"
    namespace:
      - {{labels.data_stream.namespace}}
      - default
```
- source_dataset and dataset
```yaml
# "Local" routing rules are included under this current dataset, not a special case
- source_dataset:  nginx
  # Route error logs to `nginx.error` when they're sourced from an error logfile
  - dataset: nginx.error
    if: "ctx?.file?.path?.contains('/var/log/nginx/error')"
    namespace:
      - {{labels.data_stream.namespace}}
      - default

  # Route access logs to `nginx.access` when they're sourced from an access logfile
  - dataset: nginx.access
    if: "ctx?.file?.path?.contains('/var/log/nginx/access')"
    namespace:
      - {{labels.data_stream.namespace}}
      - default
```

Reply:
Talked offline with @jsoriano 
Updated fields to be `source_dataset` and `target_dataset`, so those field names are more explicit.